runtime/src/accounts_db.rs

```diff
@@ -1298,6 +1305,35 @@ impl AccountsDb {
         }
     }
 
+    fn background_hasher(receiver: Receiver<CachedAccount>) {
+        loop {
+            let result = receiver.recv();
+            match result {
+                Ok(account) => {
+                    // if we hold the only ref, then this account doesn't need to be hashed, we ignore this account and it will disappear
+                    if Arc::strong_count(&account) > 1 {
+                        // this will cause the hash to be calculated and store inside account if it needs to be calculated
+                        let _ = (*account).hash();
```

Comment:
Point of clarification, in the PR summary it was said that: ` I don't know enough yet about the race conditions possible in the cache to attempt to update the cached entry with a new hash. `

However, this will update the hash of that cached entry no? I think it should be safe because if anything relevant to the hash is touched (lamports, account data, slot, owner, etc.), then a new account object would be stored in the cache. I.e. if for a given slot X, we had:

1) Store account A into cache
2) Background thread hashes A
3) Store account A into cache
4) Background thread commits hash for account A into its `RwLock<Hash>`

Then this should be ok because 1) and 3) stored separate versions of `A` into the cache, so the update in 4) won't update the later version of `A`, it'll update the earlier version.

Reply:
yes, that pr comment was outdated. Good catch. I will update it.